applications/nrf5340_audio/src/bluetooth/ble_core.c

```diff
@@ -267,3 +296,127 @@ int ble_core_init(ble_core_ready_t ready_callback)
+{
+	if (strcmp(input, "zero") == 0) {
+		return BLE_HCI_VSC_TX_PWR_0dBm;
+	} else if (strcmp(input, "neg1") == 0) {
+		return BLE_HCI_VSC_TX_PWR_Neg1dBm;
+	} else if (strcmp(input, "neg2") == 0) {
+		return BLE_HCI_VSC_TX_PWR_Neg2dBm;
+	} else if (strcmp(input, "neg3") == 0) {
+		return BLE_HCI_VSC_TX_PWR_Neg3dBm;
+	} else if (strcmp(input, "neg4") == 0) {
+		return BLE_HCI_VSC_TX_PWR_Neg4dBm;
+	} else if (strcmp(input, "neg5") == 0) {
+		return BLE_HCI_VSC_TX_PWR_Neg5dBm;
+	} else if (strcmp(input, "neg6") == 0) {
+		return BLE_HCI_VSC_TX_PWR_Neg6dBm;
+	} else if (strcmp(input, "neg7") == 0) {
+		return BLE_HCI_VSC_TX_PWR_Neg7dBm;
+	} else if (strcmp(input, "neg8") == 0) {
+		return BLE_HCI_VSC_TX_PWR_Neg8dBm;
+	} else if (strcmp(input, "neg12") == 0) {
+		return BLE_HCI_VSC_TX_PWR_Neg12dBm;
+	} else if (strcmp(input, "neg16") == 0) {
+		return BLE_HCI_VSC_TX_PWR_Neg16dBm;
+	} else if (strcmp(input, "neg20") == 0) {
+		return BLE_HCI_VSC_TX_PWR_Neg20dBm;
+	} else if (strcmp(input, "neg40") == 0) {
+		return BLE_HCI_VSC_TX_PWR_Neg40dBm;
+	} else {
+		return BLE_HCI_VSC_TX_PWR_INVALID;
+	}
+}
+
+static int cmd_conn_tx_pwr_set(const struct shell *shell, size_t argc, const char **argv)
+{
+	enum ble_hci_vs_tx_power tx_power;
+	struct all_conn_tx_pwr_set_info info;
+
+	if (argc != 2) {
+		shell_error(shell, "1 argument (TX power) must be provided");
+		return -EINVAL;
+	}
+
+	if (strcmp(argv[1], "help") == 0) {
+		shell_help_print(shell);
+		return 1;
+	}
+
+	tx_power = tx_power_input_check_and_enum_return(argv[1]);
+
+	if (tx_power == BLE_HCI_VSC_TX_PWR_INVALID) {
+		shell_error(shell, "Invalid TX power input");
+		return -EINVAL;
+	}
+
+	info.shell = shell;
+	info.tx_power = tx_power;
+
+	bt_conn_foreach(BT_CONN_TYPE_LE, all_conn_tx_pwr_set, &info);
+
+	shell_print(shell, "Finished setting %d dBm TX power to all connections", tx_power);
+
+	return 0;
+}
+
+static int cmd_adv_tx_pwr_set(const struct shell *shell, size_t argc, const char **argv)
+{
+	int ret;
+	enum ble_hci_vs_tx_power tx_power;
+
+	if (argc != 2) {
+		shell_error(shell, "1 argument (TX power) must be provided");
+		return -EINVAL;
+	}
+
+	if (strcmp(argv[1], "help") == 0) {
+		shell_help_print(shell);
+		return 1;
+	}
+
+	tx_power = tx_power_input_check_and_enum_return(argv[1]);
+
+	if (tx_power == BLE_HCI_VSC_TX_PWR_INVALID) {
+		shell_error(shell, "Invalid TX power input");
+		return -EINVAL;
+	}
+
+	ret = ble_hci_vsc_adv_tx_pwr_set(tx_power);
+	if (ret) {
+		shell_error(shell, "Not able to set advertisement TX power");
+	} else {
+		shell_print(shell, "%d dBm TX power set for advertisement", tx_power);
+	}
+
+	return 0;
+}
+
+SHELL_STATIC_SUBCMD_SET_CREATE(ble_core_cmd,
+			       SHELL_COND_CMD(CONFIG_SHELL, conn_tx_pwr_set, NULL,
+					      "Set connection TX power", cmd_conn_tx_pwr_set),
+			       SHELL_COND_CMD(CONFIG_SHELL, adv_tx_pwr_set, NULL,
+					      "Set advertisement TX power", cmd_adv_tx_pwr_set),
```

Comment:
What about ble_hci_vsc_pri_adv_chan_max_tx_pwr_set?

Reply:
To keep the complexity down I have not included ble_hci_vsc_pri_adv_chan_max_tx_pwr_set() for the shell commands. We can discuss if we want to include this.

Reply:
Added support for ble_hci_vsc_pri_adv_chan_max_tx_pwr_set()